docs/update_resource.md

```diff
@@ -0,0 +1,197 @@
+
+```yaml
+    # config/routes.yaml
+
+    app_book_update:
+        path: /books/{id}/edit
+        methods: [GET, PUT]
+        defaults:
+            _controller: app.controller.book:updateAction
+            _sylius:
+                form:
+                    type: app_book_custom
+                    options:
+                        validation_groups: [sylius, my_custom_group]
+```
+
+## Overriding the Criteria
+
+By default, the **updateAction** will look for the resource by id. You can easily change that criteria.
+
+```yaml
+    # config/routes.yaml
+
+    app_book_update:
+        path: /books/{title}/edit
+        methods: [GET, PUT]
+        defaults:
+            _controller: app.controller.book:updateAction
+            _sylius:
+                criteria: { title: $title }
+```
+## Custom Redirect After Success
+
+By default the controller will try to get the id of resource and redirect to the "show" route. To change that, use the following configuration.
+
+```yaml
+    # config/routes.yaml
+
+    app_book_update:
+        path: /books/{id}/edit
+        methods: [GET, PUT]
+        defaults:
+            _controller: app.controller.book:updateAction
+            _sylius:
+                redirect: app_book_index
+```
+You can also perform more complex redirects, with parameters. For example:
+
+```yaml
+    # config/routes.yaml
+
+    app_book_update:
+        path: /genre/{genreId}/books/{id}/edit
+        methods: [GET, PUT]
+        defaults:
+            _controller: app.controller.book:updateAction
+            _sylius:
+                redirect:
+                    route: app_genre_show
+                    parameters: { id: $genreId }
+```
+
+## Custom Event Name
+
+By default, there are two events dispatched during resource update, one before setting new data, the other after successful update.
+The pattern is always the same - ``{applicationName}.{resourceName}.pre/post_update``. However, you can customize the last part of the event, to provide your
+own action name.
+
+```yaml
+    # config/routes.yaml
+
+    app_book_customer_update:
+        path: /customer/book-update/{id}
+        methods: [GET, PUT]
+        defaults:
+            _controller: app.controller.book:updateAction
+            _sylius:
+                event: customer_update
+```
+This way, you can listen to ``app.book.pre_customer_update`` and ``app.book.post_customer_update`` events. It's especially useful, when you use
+``ResourceController:updateAction`` in more than one route.
+
+
+## [API] Returning resource or no content
+
+Depending on your app approach it can be useful to return a changed object or only the ``204 HTTP Code``, which indicates that everything worked smoothly.
+Sylius, by default is returning the ``204 HTTP Code``, which indicates an empty response. If you would like to receive a whole object as a response you should set a `return_content` option to true.
```

Comment:
This sentence only uses a single \` here although 2 would be consistent with the rest of the documentation. (Although one \` would have been enough.